Sources/Controllers/Settings/OSMEditing/OAMappersViewController.m

```diff
@@ -0,0 +1,398 @@
+        {
+            @try
+            {
+                _objectChanges = result[@"objectChanges"];
+                [self checkLastChanges];
+                [self generateData];
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [self.tableView reloadData];
+                });
+            }
+            @catch (NSException *e)
+            {
+                NSLog(e.reason);
+            }
+        }
+    }];
+}
+
+#pragma mark - SFSafariViewControllerDelegate
+
+- (void)safariViewControllerDidFinish:(SFSafariViewController *)controller
+{
+    [controller dismissViewControllerAnimated:YES completion:nil];
+}
+
+#pragma mark - UITableViewDataSource
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return _data.count;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return _data[section].count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    NSDictionary *item = [self getItem:indexPath];
+    UITableViewCell *outCell = nil;
+
+    NSString *type = item[@"type"];
+    if ([type isEqualToString:[OAIconTitleValueCell getCellIdentifier]])
+    {
+        OAIconTitleValueCell *cell = [tableView dequeueReusableCellWithIdentifier:[OAIconTitleValueCell getCellIdentifier]];
+        if (!cell)
+        {
+            NSArray *nib = [[NSBundle mainBundle] loadNibNamed:[OAIconTitleValueCell getCellIdentifier] owner:self options:nil];
+            cell = (OAIconTitleValueCell *) nib[0];
+            [cell showLeftIcon:NO];
+        }
+        if (cell)
+        {
+            cell.selectionStyle = [item.allKeys containsObject:@"key"] ? UITableViewCellSelectionStyleDefault : UITableViewCellSelectionStyleNone;
+
+            BOOL fullSeparator = [[_headers objectForKey:@(indexPath.section)] isEqualToString:OALocalizedString(@"shared_string_contributions")] && indexPath.row == [tableView numberOfRowsInSection:indexPath.section] - 2;
+            CGFloat leftInset = fullSeparator ? 0. : [item[@"hide_separator"] boolValue] ? CGFLOAT_MAX : 20. + [OAUtilities getLeftMargin];
+            cell.separatorInset = UIEdgeInsetsMake(0., leftInset, 0., 0.);
+
+            cell.textView.attributedText = item[@"attributed_title"];
+            cell.descriptionView.text = item[@"value"];
+            cell.descriptionView.textColor = UIColor.blackColor;
+
+            [cell showBottomDescription:[item.allKeys containsObject:@"bottom_description"]];
+            cell.bottomDescriptionView.text = item[@"bottom_description"];
+
+            NSString *rightIcon = item[@"right_icon"];
+            [cell showRightIcon:rightIcon && rightIcon.length > 0];
+            cell.rightIconView.image = [UIImage templateImageNamed:rightIcon];
+            cell.rightIconView.tintColor = item[@"tint_color"];
+        }
+        outCell = cell;
+    }
+
+    if ([outCell needsUpdateConstraints])
```

Comment:
This will be a trial for the stackview, please re-create this cell and remove if ([outCell needsUpdateConstraints])